Code/LauncherUnified/Launcher.cpp

```diff
@@ -238,9 +238,9 @@ namespace O3DELauncher
     {
         if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
         {
-            AZ::ComponentApplicationLifecycle::SignalEvent(*settingsRegistry, "CriticalAssetsCompiled", R"({})");
             // Reload the assetcatalog.xml at this point again
-            // Start Monitoring Asset changes over the network and load the AssetCatalog
+            // Start Monitoring Asset changes over the network and load the AssetCatalog.
+            // Note: When using VFS this is the first time catalog will be loaded using remote's catalog file.
```

Comment:
CriticalAssetsCompiled is supposed to be sent iff critical assets are finished compiling though, right?  Not necessarily just because we connected to AP.  Its a rare scenario where you'd fire up the runtime without compiling any assets though...

Reply:
This code is executing inside of the CompileCriticalAssets function of the Launcher which doesn't explicitly set any assets to compile critically, unlike other applications.
The Editor [CompileCriticalAssets](https://github.com/o3de/o3de/blob/development/Code/Editor/CryEdit.cpp#L1277-L1324) function waits until the AP sends back the Ready message.
The [AtomToolsApplication](https://github.com/o3de/o3de/blob/development/Gems/Atom/Tools/AtomToolsFramework/Code/Source/Application/AtomToolsApplication.cpp#L341-L389) enforces that specific assets such as ".passes", ".materials", etc.. are compiled before sedning out the CompileCriticalAssets event.